pkg/sdk/go/groups.go

```diff
@@ -0,0 +1,232 @@
+	if err != nil {
+		return "", err
+	}
+
+	resp, err := sdk.sendRequest(req, token, string(CTJSON))
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusCreated {
+		return "", errors.Wrap(ErrFailedCreation, errors.New(resp.Status))
+	}
+
+	id := strings.TrimPrefix(resp.Header.Get("Location"), fmt.Sprintf("/%s/", groupsEndpoint))
+	return id, nil
+}
+
+func (sdk mfSDK) DeleteUsersGroup(id, token string) error {
+	endpoint := fmt.Sprintf("%s/%s", groupsEndpoint, id)
+
+	url := createURL(sdk.baseURL, sdk.groupsPrefix, endpoint)
+
+	req, err := http.NewRequest(http.MethodDelete, url, nil)
+	if err != nil {
+		return err
+	}
+
+	resp, err := sdk.sendRequest(req, token, string(CTJSON))
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusNoContent {
+		return errors.Wrap(ErrFailedRemoval, errors.New(resp.Status))
+	}
+
+	return nil
+
+}
+
+func (sdk mfSDK) AssignUserToGroup(userID, groupID, token string) error {
+	endpoint := fmt.Sprintf("%s/%s/users/%s", groupsEndpoint, groupID, userID)
+	url := createURL(sdk.baseURL, sdk.groupsPrefix, endpoint)
+	req, err := http.NewRequest(http.MethodPut, url, bytes.NewReader([]byte{}))
+	if err != nil {
+		return err
+	}
+
+	resp, err := sdk.sendRequest(req, token, string(CTJSON))
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return errors.Wrap(ErrFailedConnect, errors.New(resp.Status))
+	}
+
+	return nil
+}
+
+func (sdk mfSDK) RemoveUserFromGroup(userID, groupID, token string) error {
+	endpoint := fmt.Sprintf("%s/%s/users/%s", groupsEndpoint, groupID, userID)
+	url := createURL(sdk.baseURL, sdk.groupsPrefix, endpoint)
+
+	req, err := http.NewRequest(http.MethodDelete, url, bytes.NewReader([]byte{}))
+	if err != nil {
+		return err
+	}
+	resp, err := sdk.sendRequest(req, token, string(CTJSON))
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusNoContent {
+		return errors.Wrap(ErrFailedRemoval, errors.New(resp.Status))
+	}
+
+	return nil
+}
+
+func (sdk mfSDK) ListUsersForGroup(groupID, token string, offset, limit uint64) (UsersPage, error) {
+	endpoint := fmt.Sprintf("%s/%s/users?offset=%d&limit=%d&", groupsEndpoint, groupID, offset, limit)
+	url := createURL(sdk.baseURL, sdk.groupsPrefix, endpoint)
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return UsersPage{}, err
+	}
+
+	resp, err := sdk.sendRequest(req, token, string(CTJSON))
+	if err != nil {
+		return UsersPage{}, err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return UsersPage{}, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return UsersPage{}, errors.Wrap(ErrFailedFetch, errors.New(resp.Status))
+	}
+
+	var tp UsersPage
+	if err := json.Unmarshal(body, &tp); err != nil {
+		return UsersPage{}, err
+	}
+
+	return tp, nil
+}
+
+func (sdk mfSDK) UsersGroups(token string, offset, limit uint64, id string) (UsersGroupsPage, error) {
```

Comment:
@mteodor I was thinking that you will split user groups to users and thing groups to things. Not making `pkg/sdk/go/groups`

Reply:
it was clearer to me have groups separate as a separate entity, similar to the channels.
there is a possibility that we will actually reuse these groups for things as well, maybe just adding type to the group entity.

Reply:
Its something that we should agree, I know about this idea from ~ 1 year ago :) at least. It's still not documented and you started to work in freestyle mode, not really :) But you get the point.
@mteodor I think we need new issue like Generic groups and there discuss about this and get a clear vision. It will helps a lot if we brainstorm